### PR TITLE
enhance: purge client infos periodically

### PR DESCRIPTION
**Pull request by longjiquan**
https://github.com/milvus-io/milvus/issues/31007

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -210,6 +210,7 @@ proxy:
   maxTaskNum: 1024 # max task number of proxy task queue
   connectionCheckIntervalSeconds: 120 # the interval time(in seconds) for connection manager to scan inactive client info
   connectionClientInfoTTLSeconds: 86400 # inactive client info TTL duration, in seconds
+  maxConnectionNum: 100 # the max client info numbers that proxy should manage, avoid too many client infos.
```
Comment (czs007) [inline configs/milvus.yaml:213]: too small

Comment (czs007) [reply configs/milvus.yaml:213]: at least 10000

File: internal/proxy/connection/manager.go [hunk 2]
```diff
@@ -52,11 +53,27 @@ func (s *connectionManager) checkLoop() {
...
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if s.count.Load() >= paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
+		log.Info("number of client infos exceed limit, clear them",
+			zap.Int64("num", s.count.Load()),
+			zap.Int64("limit", paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64()))
+
+		s.clientInfos.Range(func(candidate int64, info clientInfo) bool {
```
Comment (czs007) [inline internal/proxy/connection/manager.go:69]: 🔃️ 上下文的话题
No need to clear everything, just clear the excess.
Returning False in the Range indicates the termination of the Range loop.

File: internal/proxy/connection/manager.go [hunk 3]
```diff
@@ -52,11 +53,27 @@ func (s *connectionManager) checkLoop() {
...
 			s.removeLongInactiveClients()
+			// not sure if we should purge them periodically.
+			s.purgeIfNumOfClientsExceed()
 			t.Reset(paramtable.Get().ProxyCfg.ConnectionCheckIntervalSeconds.GetAsDuration(time.Second))
 		}
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if s.count.Load() >= paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
```
Comment (czs007) [inline internal/proxy/connection/manager.go:64]: We have encapsulated the Len interface for typeutil.ConcurrentMap. So, is it unnecessary to have s.count here?
If so, feel free to remove it（s.count）.

File: configs/milvus.yaml (hunk 1 above)
Comment (longjiquan) [reply configs/milvus.yaml]: changed to 10000.

File: internal/proxy/connection/manager.go (hunk 2 above)
Comment (longjiquan) [reply internal/proxy/connection/manager.go]: fixed.

File: internal/proxy/connection/manager.go (hunk 3 above)
Comment (longjiquan) [reply internal/proxy/connection/manager.go]: fixed.

File: internal/proxy/connection/manager.go [hunk 4]
```diff
@@ -52,19 +52,49 @@ func (s *connectionManager) checkLoop() {
...
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if int64(s.clientInfos.Len()) < paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
+		return
+	}
+
+	log.Info("number of client infos exceed limit, purge the oldest",
+		zap.Int64("num", int64(s.clientInfos.Len())),
+		zap.Int64("limit", paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64()))
+
+	diffNum := int64(s.clientInfos.Len()) - paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64()
```
Comment (czs007) [inline internal/proxy/connection/manager.go:71]: what if paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() changed?

File: internal/proxy/connection/manager.go [hunk 5]
```diff
@@ -52,19 +52,49 @@ func (s *connectionManager) checkLoop() {
...
 		}
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if int64(s.clientInfos.Len()) < paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
+		return
+	}
+
+	log.Info("number of client infos exceed limit, purge the oldest",
```
Comment (czs007) [inline internal/proxy/connection/manager.go:67]: how about move to bottom and also print the Duration of this purge?

File: internal/proxy/connection/manager.go (hunk 4 above)
Comment (longjiquan) [reply internal/proxy/connection/manager.go:71]: the worst case is that we purged more client infos, which i think is not very important.

File: internal/proxy/connection/manager.go (hunk 5 above)
Comment (longjiquan) [reply internal/proxy/connection/manager.go:67]: ok

**Comment by longjiquan** [comment]
rerun ut

**Pull request merged**